src/main/java/com/adroll/cantor/HLLCounter.java

```diff
@@ -490,41 +453,48 @@ public static long intersect(List<HLLCounter> hs) {
       return 0;
     }
     for (HLLCounter hll : hs) {
-      if (hll.size() == 0) {
+      if ((hll.getMinHash() == null && hll.size() == 0) || hll.getMinHash().size() == 0) {
```

Comment:
ah, so in cases where min hash isn't null we'll save on that time. that makes sense, thanks.